patroni/validator.py

```diff
@@ -0,0 +1,228 @@
+        logger.error("postgresql section is missing")
+        fatal = True
+    else:
+        if not isinstance(config["postgresql"], dict):
+            logger.warning("postgresql is not a dictionary")
+        else:
+            if "authentication" not in config["postgresql"]:
+                logger.warning("postgresql.authentication is not defined")
+            elif not isinstance(config["postgresql"]["authentication"], dict):
+                logger.warning("postgresql.authentication is not a dictionary")
+            else:
+                if "replication" not in config["postgresql"]["authentication"]:
+                    logger.warning("postgresql.authentication.replication is not defined")
+                elif not isinstance(config["postgresql"]["authentication"]["replication"], dict):
+                    logger.warning("postgresql.authentication.replication is not a dictionary")
+                else:
+                    if "username" not in config["postgresql"]["authentication"]["replication"]:
+                        logger.warning("postgresql.authentication.replication.username is not defined")
+                    elif not isinstance(config["postgresql"]["authentication"]["replication"]["username"], str):
+                        logger.warning("postgresql.authentication.replication.username is not a string")
+                    if "password" not in config["postgresql"]["authentication"]["replication"]:
+                        logger.warning("postgresql.authentication.replication.password is not defined")
+                    elif not isinstance(config["postgresql"]["authentication"]["replication"]["password"], str):
+                        logger.warning("postgresql.authentication.replication.password is not a string")
+                if "superuser" not in config["postgresql"]["authentication"]:
+                    logger.warning("postgresql.authentication.superuser is not defined")
+                elif not isinstance(config["postgresql"]["authentication"]["superuser"], dict):
+                    logger.warning("postgresql.authentication.superuser is not a dictionary")
+                else:
+                    if "username" not in config["postgresql"]["authentication"]["superuser"]:
+                        logger.warning("postgresql.authentication.superuser.username is not defined")
+                    elif not isinstance(config["postgresql"]["authentication"]["superuser"]["username"], str):
+                        logger.warning("postgresql.authentication.superuser.username is not a string")
+                    if "password" not in config["postgresql"]["authentication"]["superuser"]:
+                        logger.warning("postgresql.authentication.superuser.password is not defined")
+                    elif not isinstance(config["postgresql"]["authentication"]["superuser"]["password"], str):
+                        logger.warning("postgresql.authentication.superuser.password is not a string")
```

Comment:
Why not do a couple of loops instead?
There is also auth parameters section for `rewind`.
Lack of password is actually fine, people might use different authentication methods like kerberos or client certificates.